network/p2p/blob_service.go

```diff
@@ -158,3 +171,33 @@ func (s *blobServiceSession) GetBlob(ctx context.Context, c cid.Cid) (blobs.Blob
 func (s *blobServiceSession) GetBlobs(ctx context.Context, ks []cid.Cid) <-chan blobs.Blob {
 	return s.session.GetBlocks(ctx, ks)
 }
+
+type rateLimitedBlockStore struct {
+	blockstore.Blockstore
+	limiter *rate.Limiter
+	metrics module.RateLimitedBlockstoreMetrics
+}
+
+func newRateLimitedBlockStore(bs blockstore.Blockstore, r float64, b int) *rateLimitedBlockStore {
+	return &rateLimitedBlockStore{
+		Blockstore: bs,
+		limiter:    rate.NewLimiter(rate.Limit(r), b),
+		metrics:    metrics.NewRateLimitedBlockstoreCollector(),
+	}
+}
+
+func (r *rateLimitedBlockStore) Get(ctx context.Context, c cid.Cid) (blocks.Block, error) {
+	size, err := r.Blockstore.GetSize(ctx, c)
+	if err != nil {
+		return nil, err
+	}
+
+	err = r.limiter.WaitN(ctx, size)
```

Comment:
`WaitN`'s godocs says
> WaitN blocks until lim permits n events to happen. It returns an error if n exceeds the Limiter's burst size, the Context is canceled, or the expected wait time exceeds the Context's Deadline. The burst limit is ignored if the rate limit is Inf.

I think this means that the burst limit must be >=max blob size. Can you add a warning in the godocs for `WithRateLimit`